<launch>
  <!-- turtlebot_teleop_key already has its own built in velocity smoother -->
  <node pkg="turtlebot3_interactions" type="turtlebot3_marker_server" name="turtlebot3_marker_server">
     <remap from="interactive_marker_velocity_smoother/raw_cmd_vel" to="cmd_vel"/>
  </node>

 <!-- velocity smoother for the interactive marker server output -->
  <node pkg="nodelet" type="nodelet" name="interactive_marker_velocity_smoother"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet /mobile_base_nodelet_manager">
    <rosparam file="$(find turtlebot3_bringup)/param/smoother.yaml" command="load"/>
    <remap from="interactive_marker_velocity_smoother/odometry" to="/odom"/>
    <remap from="interactive_marker_velocity_smoother/robot_cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="interactive_marker_velocity_smoother/smooth_cmd_vel" to="/cmd_vel_mux/input/teleop"/>
  </node>
</launch>
